src/routes/render/renderAuthenticatedImage.ts

```diff
@@ -0,0 +1,111 @@
+import { getConfig } from '../../utils/config'
+import { GenericStorageBackend } from '../../backend/generic'
+import { FileBackend } from '../../backend/file'
+import { S3Backend } from '../../backend/s3'
+import { FromSchema } from 'json-schema-to-ts'
+import { FastifyInstance } from 'fastify'
+import { Obj } from '../../types/types'
+import { normalizeContentType, transformPostgrestError } from '../../utils'
+import { Imgproxy } from '../../renderer/imgproxy'
+import { AxiosError } from 'axios'
+
+const { region, globalS3Bucket, globalS3Endpoint, storageBackendType, imgProxyURL } = getConfig()
+
+let storageBackend: GenericStorageBackend
+
+if (storageBackendType === 'file') {
+  storageBackend = new FileBackend()
+} else {
+  storageBackend = new S3Backend(region, globalS3Endpoint)
+}
+
+const imageRenderer = new Imgproxy(storageBackend, {
+  url: imgProxyURL || '',
+})
+
+const renderAuthenticatedImageParamsSchema = {
+  type: 'object',
+  properties: {
+    bucketName: { type: 'string', examples: ['avatars'] },
+    '*': { type: 'string', examples: ['folder/cat.png'] },
+  },
+  required: ['bucketName', '*'],
+} as const
+
+const renderImageQuerySchema = {
+  type: 'object',
+  properties: {
+    height: { type: 'number', examples: [100] },
+    width: { type: 'number', examples: [100] },
+    resize: { type: 'string', enum: ['fill', 'fit', 'fill-down', 'force', 'auto'] },
+  },
+} as const
+
+interface renderImageRequestInterface {
+  Params: FromSchema<typeof renderAuthenticatedImageParamsSchema>
+  Querystring: FromSchema<typeof renderImageQuerySchema>
+}
+
+export default async function routes(fastify: FastifyInstance) {
+  const summary = 'Render an authenticated image with the given transformations'
+  fastify.get<renderImageRequestInterface>(
+    '/authenticated/:bucketName/*',
+    {
+      schema: {
+        params: renderAuthenticatedImageParamsSchema,
+        summary,
+        response: { '4xx': { $ref: 'errorSchema#', description: 'Error response' } },
+        tags: ['object'],
+      },
+    },
+    async (request, response) => {
+      const { bucketName } = request.params
+      const objectName = request.params['*']
+
+      const objectResponse = await request.postgrest
+        .from<Obj>('objects')
+        .select('id')
+        .match({
+          name: objectName,
+          bucket_id: bucketName,
+        })
+        .single()
+
+      if (objectResponse.error) {
+        const { status, error } = objectResponse
+        request.log.error({ error }, 'error object')
+        return response.status(400).send(transformPostgrestError(error, status))
+      }
+
+      const s3Key = `${request.tenantId}/${bucketName}/${objectName}`
+
+      try {
+        const imageResponse = await imageRenderer.transform(globalS3Bucket, s3Key, request.query)
```

Comment:
is there a way to stream the response as a stream [here](https://github.com/supabase/storage-api/pull/199/files#diff-db92043d530d0a7fd1b0b47bf588452fef1c11d5eba52614e9768f9d7d556e59R49) and send that stream directly to the client instead of doing an await?